adapters/base/src/main/java/org/apache/cassandra/sidecar/adapters/base/TokenRangeReplicaProvider.java

```diff
@@ -285,7 +285,9 @@ String of(String endpoint)
                 {
                     LOGGER.debug("Found gossipInfoEntry={}", gossipInfoEntry);
                     String hostStatus = gossipInfoEntry.status();
-                    if (hostStatus != null && hostStatus.startsWith("BOOT_REPLACE,"))
+                    String hostStatusWithPort = gossipInfoEntry.statusWithPort();
+                    if ((hostStatus != null && hostStatus.startsWith("BOOT_REPLACE,")) ||
+                        (hostStatusWithPort != null && hostStatusWithPort.startsWith("BOOT_REPLACE,")))
```

Comment:
nit: it might be worth to add a warning when `STATUS` field is missing when cluster runs in 4.x, since it marks a potential gossip bug. It is not the sidecar's responsibility to monitor it, so I am fine without it too. I would defer to your decision.

Reply:
Oops, seeing this late. I agree with you that we could do without the log line, since it really does not affect the sidecar or this APIs behavior.